### PR TITLE
Volinga model and NVOL documentation

### PR DESCRIPTION
**Pull request by Frivas97**
This Pull Request implements Volinga model, which is a simple adaptation of nerfacto for real-time. It also add the documentation for the conversion of ckpts into nvols that can be used in Unreal Engine.

Finally, this PR also changes some attributes of nerfacto from parameters to buffers. Since they are not being optimized, it is a better practice to use buffers (https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723/2).

### CODE REVIEW

**Comment by tancik** [comment]
Here is the readthedocs warning
`checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/plenoptix-nerfstudio/checkouts/1486/docs/quickstart/unreal_engine.md: WARNING: document isn't included in any toctree`

You can compile the docs locally with `python scripts/docs/build_docs.py --clean-cache`
Then run `ns-dev-test`
If both pass, then it should compile on readthedocs.

File: nerfstudio/models/nerfacto.py [hunk 1]
```diff
@@ -93,8 +99,74 @@ class NerfactoModelConfig(ModelConfig):
...
+    proposal_weights_anneal_max_num_iters: int = 1000
+    """Max num iterations for the annealing function."""
+    use_single_jitter: bool = True
+    """Whether use single jitter or not for the proposal networks."""
+    predict_normals: bool = False
+    """Whether to predict normals or not."""
+
+
+@dataclass
+class VolingaModelConfig(ModelConfig):
```
Comment (tancik) [inline nerfstudio/models/nerfacto.py:132]: Is there a reason a new model config is necessary? It looks like it just has different arguments, can these be applied in the method_configs definition?

File: nerfstudio/models/nerfacto.py [hunk 2]
```diff
@@ -93,8 +99,74 @@ class NerfactoModelConfig(ModelConfig):
...
+    far_plane: float = 1000.0
+    """How far along the ray to stop sampling."""
+    background_color: Literal["random", "last_sample"] = "last_sample"
+    """Whether to randomize the background color."""
+    hidden_dim = 32
+    """Dimension of hidden layers"""
+    hidden_dim_color = 32
+    """Dimension of hidden layers for color network"""
+    hidden_dim_transient = 32
+    """Dimension of hidden layers for transient network"""
```
Comment (tancik) [inline nerfstudio/models/nerfacto.py:147]: These can added to the nerfacto config.

File: docs/quickstart/unreal_engine.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Exporting your model to NVOL for Unreal Engine
```
Comment (tancik) [inline docs/quickstart/unreal_engine.md:1]: I think I'm going to create a new section on the docs called `Extensions` that will can include this and others like Blender and ROS. I'll try to push an update today

File: Dockerfile [hunk 4]
```diff
@@ -140,4 +140,3 @@ WORKDIR /workspace
 
 # Install nerfstudio cli auto completion and enter shell if no command was provided.
 CMD ns-install-cli --mode install && /bin/bash
-
```
Comment (tancik) [inline Dockerfile:143]: Change not needed.

File: docs/quickstart/unreal_engine.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <img width="800" alt="image" src="./imgs/desolation_unreal.png">
+</p>
+
+
+## Overview
+
+NVOL is a new standard file format to store NeRFs in a fast and efficient way. NVOL files can be obtained from NeRFStudio checkpoints files (.ckpt) using Volinga Suite (https://volinga.ai/).
+
+
+> ⚠️ Volinga Suite is not yet publicly available. Sign up for the beta at https://volinga.ai/ to get access.
```
Comment (tancik) [inline docs/quickstart/unreal_engine.md:13]: Use an admonition

**Review by tancik — COMMENTED** [review]
Looks cool!

The `export_nvol.png` has a typo for `Desolation`

File: nerfstudio/models/nerfacto.py (hunk 2 above)
Comment (Frivas97) [reply nerfstudio/models/nerfacto.py:147]: Do you mean modifying this in the original nerfacto model?

File: docs/quickstart/unreal_engine.md (hunk 3 above)
Comment (Frivas97) [reply docs/quickstart/unreal_engine.md:1]: Nice, should I wait for you to complete the PR then?

File: nerfstudio/configs/method_configs.py [hunk 6]
```diff
@@ -137,6 +138,37 @@
...
+    pipeline=VanillaPipelineConfig(
+        datamanager=VanillaDataManagerConfig(
+            dataparser=NerfstudioDataParserConfig(),
+            train_num_rays_per_batch=4096,
+            eval_num_rays_per_batch=4096,
+            camera_optimizer=CameraOptimizerConfig(
+                mode="SO3xR3", optimizer=AdamOptimizerConfig(lr=6e-4, eps=1e-8, weight_decay=1e-2)
+            ),
+        ),
+        model=VolingaModelConfig(eval_num_rays_per_chunk=1 << 15),
```
Comment (machenmusik) [inline nerfstudio/configs/method_configs.py:156]: Looks as though the Volinga model config uses Nerfacto model with specific parameters?  If so, may be cleaner to use existing NerfactoModelConfig with desired Volinga settings, similar to how instant-ngp-bounded leverages instant-ngp.

**Comment by machenmusik** [comment]
(How does one test whether it works?  Initially the assumption is that it does at the moment, but one might expect additional changes to nerfacto.)

File: docs/quickstart/unreal_engine.md (hunk 3 above)
Comment (tancik) [reply docs/quickstart/unreal_engine.md:1]: https://github.com/nerfstudio-project/nerfstudio/pull/1488

File: nerfstudio/models/nerfacto.py (hunk 2 above)
Comment (tancik) [reply nerfstudio/models/nerfacto.py:147]: Add these to the nerfacto config but with the default nerfacto values.

Comment (Frivas97) [reply nerfstudio/models/nerfacto.py:147]: They were already added (lines 76 yo 81 of nerfacto.py)

File: nerfstudio/models/nerfacto.py (hunk 1 above)
Comment (Frivas97) [reply nerfstudio/models/nerfacto.py]: You are completely right. I have changed Volinga to use the existing NerfactoModelConfig

File: docs/quickstart/unreal_engine.md (hunk 3 above)
Comment (Frivas97) [reply docs/quickstart/unreal_engine.md]: Done

File: docs/quickstart/unreal_engine.md (hunk 5 above)
Comment (Frivas97) [reply docs/quickstart/unreal_engine.md]: Done

File: docs/index.md [hunk 7]
```diff
@@ -58,8 +58,8 @@ Contributing<reference/contributing>
 ```{toctree}
 :hidden:
 :caption: Extensions
-
 extensions/blender_addon
+extensions/unreal_engine
```
Comment (tancik) [inline docs/index.md:62]: Lets use a shorter label for the sidebar. perhaps,
`Unreal Engine<extensions/unreal_engine>`

File: docs/extensions/unreal_engine.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# Exporting your model to NVOL for Unreal Engine
+
+<p align="center">
+    <img width="800" alt="image" src="./imgs/desolation_unreal.png">
+</p>
+
+
+## Overview
+
+NVOL is a new standard file format to store NeRFs in a fast and efficient way. NVOL files can be obtained from NeRFStudio checkpoints files (.ckpt) using Volinga Suite (https://volinga.ai/).
```
Comment (tancik) [inline docs/extensions/unreal_engine.md:10]: `NVOL is a file format to store NeRFs in a fast and efficient way. NVOL files can be obtained from NeRFStudio checkpoints files (.ckpt) using the [Volinga Suite](https://volinga.ai/)`

File: docs/extensions/unreal_engine.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+
+!!! warning Volinga Suite is not yet publicly available
+    If you are interested in converting your models to NVOL, sign up for the beta at https://volinga.ai/.
```
Comment (tancik) [inline docs/extensions/unreal_engine.md:14]: -> `[volinga.ai](https://volinga.ai)`

File: docs/extensions/unreal_engine.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Exporting your model to NVOL for Unreal Engine
+
+<p align="center">
+    <img width="800" alt="image" src="./imgs/desolation_unreal.png">
+</p>
```
Comment (tancik) [inline docs/extensions/unreal_engine.md:5]: For images use this format
https://github.com/nerfstudio-project/nerfstudio/blob/1a2bbc5725261f617604de8d066cc1a8f33b6316/docs/quickstart/custom_dataset.md?plain=1#L158-L162

File: docs/extensions/unreal_engine.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+
+## Overview
+
+NVOL is a new standard file format to store NeRFs in a fast and efficient way. NVOL files can be obtained from NeRFStudio checkpoints files (.ckpt) using Volinga Suite (https://volinga.ai/).
+
+
+!!! warning Volinga Suite is not yet publicly available
```
Comment (tancik) [inline docs/extensions/unreal_engine.md:13]: For admonitions, use the following format

```
:::{admonition} Note
:class: warning
Volinga Suite is not yet...
:::
```

**Review by tancik — COMMENTED** [review]
Left a few comments, the current docs are a bit broken 
image

I recommend building them locally to make sure they look as expected.

**Review by tancik — COMMENTED** [review]
Looks good now, just remove the change to the docker file.

**Review by tancik — APPROVED** [review]
Lgtm. I'm excited to see what people do with unreal engine!

**Review by tancik — APPROVED** [review]

**Pull request merged**